stock_picking_group_by_partner_by_carrier/models/procurement_group.py

```diff
@@ -4,7 +4,6 @@
 class ProcurementGroup(models.Model):
     _inherit = "procurement.group"
 
-    carrier_id = fields.Many2one("delivery.carrier", string="Delivery Method")
```

Comment:
We need a migration script for this field to move its metadata to the new addon otherwise the upgrade of this module will drop the column.

Reply:
@jbaudoux if you want I'll take care of this, give me access to your fork so I can push the fix directly

Reply:
@sebalix If you're maintainer, you should be able. Otherwise, make a PR on this and I'll include it

Reply:
@sebalix Is this https://github.com/OCA/stock-logistics-workflow/pull/936/commits/2d1e2590e7a8a26b329fccb40791004e3feefc33 solving the issue?

Reply:
@jbaudoux no, because if `stock_picking_group_by_partner_by_carrier` is upgraded before, Odoo will drop the database column post-upgrade so you will lose the data. You need to move your script in pre-migration (in `stock_picking_group_by_partner_by_carrier/migrations/`) even if the `delivery_procurement_group_carrier` doesn't exist in the `ir_module_module` table, that doesn't matter (based on my tests...). Then when the `delivery_procurement_group_carrier` addon will be installed Odoo won't re-create the column as it hasn't been drop (still referenced by a module in `ir_model_data`), so the data will be preserved.

Reply:
@i-vyshnevska I didn't see you move the script in `pre-migration`, but this would need to be in the `stock_picking_group_by_partner_by_carrier` addon instead

Reply:
@sebalix we don't need to provide migration for stock_picking_group_by_partner_by_carrier, more of it we have errors when trying to move fields with the script as data already was created 
no migration scrips needed

Reply:
I agree, no migration needed at time, my bad.

Reply:
@jbaudoux can you remove the migration script then? As `stock_picking_group_by...` depends on the current one, the migration path is already good, no need to do anything.

Reply:
Sorry a little late to the party. But if we are talking about this one `delivery_procurement_group_carrier/migrations/14.0.1.0.0/pre-migration.py`
I think it make sense, no ?

Reply:
@sebalix Is it your final answer....

Reply:
migrations/14.0.1.0.0/pre-migration.py removed

Reply:
@TDu That what I thought initially, but after a test it is not needed. Indeed, this new addon is a dependency of the existing one, so Odoo will install this new addon first, which add a line in `ir_model_data` that references the `carrier_id` field, then the `stock_group_by...` addon is updated where the `carrier_id` isn't declared anymore so its reference is removed from `ir_model_data`, but as there is still a reference from `delivery_procurement_group_carrier` addon in `ir_model_data` then Odoo doesn't drop the column.

@jbaudoux thank you! Sorry for the noise...